src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs

```diff
@@ -1073,6 +1073,28 @@ public static string GetMachineFQDN(string hostname)
             return fqdn.ToString();
         }
 
+        public static bool IsIntegratedSecurity()
+        {
+            SqlConnectionStringBuilder builder = new(DataTestUtility.TCPConnectionString);
+            return builder.IntegratedSecurity;  
+        }
+
+        public static bool IsNotLocalhost()
+        {
+            // get the tcp connection string
+            SqlConnectionStringBuilder builder = new(DataTestUtility.TCPConnectionString);
+
+            // remove tcp from connection string
+            string dataSourceStr = builder.DataSource.Replace("tcp:", "");
+            // create a string array
+            string[] serverNamePartsByBackSlash = dataSourceStr.Split('\\');
+            // first element of array is the hostname
+            string hostname = serverNamePartsByBackSlash[0];
+
+            // check if hostname = localhost
+            return !hostname.Equals("localhost", StringComparison.OrdinalIgnoreCase);
```

Comment:
`localhost` is not the only convention, `.` has the same meaning if you're not exactly looking for the same name.

Reply:
Now checks if hostname = localhost or . or 127.0.0.1 or ::1